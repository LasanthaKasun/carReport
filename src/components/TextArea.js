import React from "react";
import Error from "./Error";

const TextArea = React.forwardRef(
  ({ label, isRequired, onChangeText, error }, ref) => (
    <div className="mt-6">
      <label className="block mb-2 text-sm font-medium text-gray-900 dark:text-gray-400">
        {label}
        {isRequired && <span className="text-red-600">*</span>}
      </label>
      <textarea
        ref={ref}
        onChange={onChangeText}
        rows="4"
        className="block p-2.5 w-full h-72 text-sm text-gray-900 bg-gray-50 rounded-lg border border-gray-300 focus:ring-blue-500 focus:border-blue-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
        placeholder="Your message..."
      ></textarea>
      {error && <Error message={error.message} />}
    </div>
  )
);
export default TextArea;
