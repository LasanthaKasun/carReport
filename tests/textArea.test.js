import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import TextArea from "../src/components/TextArea";

it("should render text area component", () => {
  const component = renderer.create(<TextArea />);
  expect(component).toMatchSnapshot();
});

it("should render text area with props", () => {
  render(
    <TextArea label="" isRequired={true} onChangeText={null} error={""} />
  );
});

it("should render text area with error", () => {
  render(
    <TextArea label="" isRequired={true} onChangeText={null} error={"error"} />
  );
});
