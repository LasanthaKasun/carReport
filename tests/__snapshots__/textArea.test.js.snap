// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render text area component 1`] = `
<div
  className="mt-6"
>
  <label
    className="block mb-2 text-sm font-medium text-gray-900 dark:text-gray-400"
  />
  <textarea
    className="block p-2.5 w-full h-72 text-sm text-gray-900 bg-gray-50 rounded-lg border border-gray-300 focus:ring-blue-500 focus:border-blue-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
    placeholder="Your message..."
    rows="4"
  />
</div>
`;
